Core/src/IfcGeometryConverter/PlacementConverter.h

```diff
@@ -116,14 +117,12 @@ namespace OpenInfraPlatform {
                     /*! \brief Converts \c IfcDirection to a vector.
 
                     \param[in]	ifcDirection	IfcDirection entity to be interpreted.
-                    \param[out] direction		Calculated 2D or 3D vector.
+                    \returns direction		    Calculated 2D or 3D vector.
 
                     \note The direction is normalized.
                     \note The default value for the direction needs to be set outside the function.
                     */
-                    void convertIfcDirection(
-                        const std::shared_ptr<typename IfcEntityTypesT::IfcDirection>& ifcDirection,
-                        carve::geom::vector<3>& direction)
+                    carve::geom::vector<3> convertIfcDirection(const EXPRESSReference<typename IfcEntityTypesT::IfcDirection>& ifcDirection)

```

Comment:
This should be done everywhere. Change std::shared_ptr<> with EXPRESSReference<>